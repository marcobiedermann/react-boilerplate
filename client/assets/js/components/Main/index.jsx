import PropTypes from 'prop-types';
import React from 'react';

const Main = props => {
  const { children } = props;

  return <main>{children}</main>;
};

Main.propTypes = {
  children: PropTypes.node,
};

Main.defaultProps = {
  children: null,
};

export default Main;
