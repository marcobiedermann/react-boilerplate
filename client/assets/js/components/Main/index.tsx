import React from 'react';

const Main: React.FC = (props) => {
  const { children } = props;

  return <main>{children}</main>;
};

export default Main;
