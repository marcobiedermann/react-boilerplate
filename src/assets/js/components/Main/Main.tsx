import React, { ReactNode } from 'react';

export interface MainProps {
  children: ReactNode;
}

function Main(props: MainProps): JSX.Element {
  return <main {...props} />;
}

export type { MainProps };
export default Main;
