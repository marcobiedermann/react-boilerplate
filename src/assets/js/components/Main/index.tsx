import React, { ReactNode } from 'react';

export interface MainProps {
  children: ReactNode;
}

function Main(props: MainProps): JSX.Element {
  return <main {...props} />;
}

export default Main;
