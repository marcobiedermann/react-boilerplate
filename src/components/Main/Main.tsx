import { ReactNode } from 'react';

interface MainProps {
  children: ReactNode;
}

function Main(props: MainProps): JSX.Element {
  return <main {...props} />;
}

export type { MainProps };
export default Main;
