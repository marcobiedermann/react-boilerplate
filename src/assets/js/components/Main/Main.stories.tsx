import type { Meta, StoryObj } from '@storybook/react';
import Main from '.';

const meta: Meta<typeof Main> = {
  component: Main,
};

type Story = StoryObj<typeof Main>;

const Default: Story = {
  args: {
    children: 'Main',
  },
};

export { Default };
export default meta;
