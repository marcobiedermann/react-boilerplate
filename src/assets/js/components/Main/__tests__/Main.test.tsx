import { composeStories } from '@storybook/testing-react';
import { render } from '@testing-library/react';
import * as stories from '../Main.stories';

const { Default } = composeStories(stories);

describe('main', () => {
  it('should render correctly', () => {
    const { container } = render(<Default />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
