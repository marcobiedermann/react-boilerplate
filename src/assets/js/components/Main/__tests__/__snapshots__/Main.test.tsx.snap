// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`main > should render correctly 1`] = `
<main>
  Main
</main>
`;
